hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -753,10 +755,47 @@ protected void checkPath(Path path) {
           return;
       }
     }
+
+    // The authority of the input path does not match with the
+    // authority of the client. Check if it matches with any
+    // of the available NNs in the system.
+    String nns = getConf().get("dfs.namenodes.rpc.addresses");
```

Comment:
I'm not sure reading the list of NN from the configuration will work in the case described by Fabio. I think this list will contain only consul address (and maybe even only one) and not the IPs that Fabio got.

Reply:
You are right. I have changed the code to get the list of NNs from a alive NN and then compare both the IP and CanonicalHostName  .